### PR TITLE
Fix ISIS Reflectometry batch load crash

### LINKED ISSUE #27178: ISIS Reflectometry GUI causes mantid crash on *.tbl load attempt
### Expected behavior
The ISIS Reflectometry GUI should not cause a crash if a user attempts to load an unsupported file type.

### Actual behavior
The GUI crashes mantid

### Steps to reproduce the behavior
- Open ISIS Reflectometry
- `Batch -> Load`
- Attempt to lad `INTER_NR_test2.tbl` from `SampleData-ISIS`
- Observe crash

### Platforms affected
All

### PR DESCRIPTION
**Pull request by giovannidisiena**
~~Merge after #27233~~

**Description of work.**
This PR adds a try/catch block to handle runtime errors which result in a hard crash when attempting to load an unsupported file type/format into the ISIS Reflectometry GUI through Batch -> Load

**To test:**

- Open ISIS Reflectometry GUI
- Load in any file from SampleData-ISIS
- Observe handled exception
----
- Reopen GUI
- Enter run number `13460`, angle `0.5`
- Batch -> Save
- Close/Reopen GUI
- Load saved file
----
- Close/Reopen GUI
- Edit saved file in text editor (mess with the formatting etc)
- Attempt to load file
- Observe handled exception

Fixes #27178

*This does not require release notes* because **it is a bug fix which is unlikely to have been seen by users**

---

#### Reviewer ####

Please comment on the following ([full description](http://developer.mantidproject.org/ReviewingAPullRequest.html)):

##### Code Review #####

- Is the code of an acceptable quality?
- Does the code conform to the [coding standards](http://developer.mantidproject.org/Standards/)?
- Are the unit tests small and test the class in isolation?
- If there are changes in the release notes then do they describe the changes appropriately?

##### Functional Tests #####

- Do changes function as described? Add comments below that describe the tests performed?
- Do the changes handle unexpected situations, e.g. bad input?
- Has the relevant (user and developer) documentation been added/updated?

Does everything look good? Mark the review as **Approve**. A member of `@mantidproject/gatekeepers` will take care of it.

### CODE REVIEW

**Review by Harrietbrown — DISMISSED** [review]
This handles the exception well, a simple fix.

Good work

File: qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp [hunk 1]
```diff
@@ -195,7 +196,15 @@ void MainWindowPresenter::notifyLoadBatchRequested(int tabIndex) {
   auto filename = QFileDialog::getOpenFileName();
   if (filename == "")
     return;
-  auto map = MantidQt::API::loadJSONFromFile(filename);
+  QMap<QString, QVariant> map;
+  try {
+    map = MantidQt::API::loadJSONFromFile(filename);
+  } catch (const std::runtime_error &e) {
+    QMessageBox msgBox;
```
Comment (DanNixon) [inline qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp]: `QMessageBox` has [static functions](https://doc.qt.io/qt-5/qmessagebox.html#warning) that make this simpler.

**Review by DanNixon — CHANGES_REQUESTED** [review]

**Comment by giovannidisiena** [comment]
Merge after #27233

File: qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp [hunk 2]
```diff
@@ -195,7 +196,15 @@ void MainWindowPresenter::notifyLoadBatchRequested(int tabIndex) {
   auto filename = QFileDialog::getOpenFileName();
   if (filename == "")
     return;
-  auto map = MantidQt::API::loadJSONFromFile(filename);
+  QMap<QString, QVariant> map;
+  try {
+    map = MantidQt::API::loadJSONFromFile(filename);
+  } catch (const std::runtime_error &e) {
+    QMessageBox::warning(
```
Comment (DanNixon) [inline qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp]: I just realised this is the presenter :man_facepalming:.

Following the MVP pattern strictly your view should have a function to show the message box with a string parameter and the presenter would call this.

Comment (giovannidisiena) [reply qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp]: Oops 🤦‍♂

File: qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp [hunk 3]
```diff
@@ -20,6 +20,7 @@
 #include "Reduction/Batch.h"
 
 #include <QFileDialog>
+#include <QMessageBox>
```
Comment (gemmaguest) [inline qt/scientific_interfaces/ISISReflectometry/GUI/MainWindow/MainWindowPresenter.cpp]: I don't think this is required?

```suggestion
```

**Comment by gemmaguest** [comment]
Thanks for removing the direct use of QMessageBox. There is some other use of Qt types in the presenter but that's not related to this PR. I've raised issue #27242 to deal with it.

**Comment by gemmaguest** [comment]
@DanNixon The dependency has been merged so I think this is ready for another look now

**Review by DanNixon — APPROVED** [review]

**Pull request merged**